tests/etch.rs

```diff
@@ -0,0 +1,370 @@
+      .rpc_server(&rpc_server)
+      .run_and_deserialize_output::<ord::subcommand::wallet::balance::Output>();
+
+  assert_eq!(output.cardinal, 5000000000);
+
+  CommandBuilder::new("--regtest wallet inscribe --fee-rate 0 --file foo.txt --postage 50btc")
+    .write("foo.txt", "FOO")
+    .rpc_server(&rpc_server)
+    .run_and_deserialize_output::<Inscribe>();
+
+  rpc_server.mine_blocks_with_subsidy(1, 0);
+
+  let output =
+    CommandBuilder::new("--regtest --index-runes-pre-alpha-i-agree-to-get-rekt wallet balance")
+      .rpc_server(&rpc_server)
+      .run_and_deserialize_output::<ord::subcommand::wallet::balance::Output>();
+
+  assert_eq!(output.cardinal, 0);
+
+  CommandBuilder::new(
+    format!(
+    "--index-runes-pre-alpha-i-agree-to-get-rekt --regtest wallet etch --rune {} --divisibility 1 --fee-rate 1 --supply 1000 --symbol ¢",
+    Rune(RUNE),
+  ))
+  .rpc_server(&rpc_server)
+  .stderr_regex("error: JSON-RPC error: .*")
+  .expected_exit_code(1)
+  .run_and_extract_stdout();
+
+  rpc_server.mine_blocks(1);
+}
+
+#[test]
+fn inscribe_does_not_select_runic_utxos() {
+  let rpc_server = test_bitcoincore_rpc::builder()
+    .network(Network::Regtest)
+    .build();
+
+  create_wallet(&rpc_server);
+
+  rpc_server.mine_blocks_with_subsidy(1, 10000);
+
+  CommandBuilder::new(
+    format!(
+    "--index-runes-pre-alpha-i-agree-to-get-rekt --regtest wallet etch --rune {} --divisibility 1 --fee-rate 0 --supply 1000 --symbol ¢",
+    Rune(RUNE),
+  ))
+  .rpc_server(&rpc_server)
+  .run_and_deserialize_output::<Output>();
+
+  rpc_server.mine_blocks_with_subsidy(1, 0);
+
+  let output =
+    CommandBuilder::new("--regtest --index-runes-pre-alpha-i-agree-to-get-rekt wallet balance")
+      .rpc_server(&rpc_server)
+      .run_and_deserialize_output::<ord::subcommand::wallet::balance::Output>();
+
+  assert_eq!(output.cardinal, 0);
+  assert_eq!(output.ordinal, 0);
+  assert_eq!(output.runic, 10000);
+
+  CommandBuilder::new("--regtest --index-runes-pre-alpha-i-agree-to-get-rekt wallet inscribe --fee-rate 0 --file foo.txt")
+    .write("foo.txt", "FOO")
+    .rpc_server(&rpc_server)
+    .expected_exit_code(1)
+    .expected_stderr("error: wallet contains no cardinal utxos\n")
+    .run_and_extract_stdout();
+}
+
+#[test]
+fn send_amount_does_not_select_runic_utxos() {
+  let rpc_server = test_bitcoincore_rpc::builder()
+    .network(Network::Regtest)
+    .build();
+
+  create_wallet(&rpc_server);
+
+  rpc_server.mine_blocks_with_subsidy(1, 10000);
+
+  CommandBuilder::new(
+    format!(
+    "--index-runes-pre-alpha-i-agree-to-get-rekt --regtest wallet etch --rune {} --divisibility 1 --fee-rate 0 --supply 1000 --symbol ¢",
+    Rune(RUNE),
+  ))
+  .rpc_server(&rpc_server)
+  .run_and_deserialize_output::<Output>();
+
+  rpc_server.mine_blocks_with_subsidy(1, 0);
+
+  CommandBuilder::new("--regtest --index-runes-pre-alpha-i-agree-to-get-rekt wallet send --fee-rate 1 bcrt1qs758ursh4q9z627kt3pp5yysm78ddny6txaqgw 1sat")
```

Comment:
increase to an amount above dust limit to make sure that the correct error is triggered